### ISSUE TITLE
PHP 5.6 fails to build with GCC 10

### ISSUE THREAD

**Issue by anomiex** [oerdnj/deb.sury.org#1441]
**Frequently asked questions**

- [x] I have read [Frequenty Asked Questions](https://github.com/oerdnj/deb.sury.org/wiki/Frequently-Asked-Questions)

**Describe the bug**
The source package php5.6 version 5.6.40-29+0~20200514.35+debian10~1.gbpcc49a4 fails to link with gcc 10; it compiles and links fine with gcc 9.

This appears to be due to the "Default to -fno-common" change described at https://gcc.gnu.org/gcc-10/porting_to.html

**To Reproduce**

With gcc 10 as the default gcc, fetch the source and attempt to compile it using `dpkg-buildpackage -b -uc`.

**Expected behavior**

It compiles and links successfully, and produces .deb files that can be installed.

**Observed behavior**

Linking fails with the following result:
```
libtool: link: x86_64-linux-gnu-gcc -I/usr/include -g -O2 -fdebug-prefix-map=/usr/local/src/bugfix-debs/php/php5.6/php5.6-5.6.40=. -fstack-protector-strong -Wformat -Werror=format-security -O2 -Wall -pedantic -fsigned-char -fno-strict-aliasing -g -Wl,-z -Wl,relro -Wl,-z -Wl,now -Wl,--as-needed Zend/.libs/zend_dtrace.d.o ext/date/.libs/php_date.o ext/date/lib/.libs/astro.o ext/date/lib/.libs/dow.o ext/date/lib/.libs/parse_date.o ext/date/lib/.libs/parse_tz.o ext/date/lib/.libs/timelib.o ext/date/lib/.libs/tm2unixtime.o ext/date/lib/.libs/unixtime2tm.o ext/date/lib/.libs/parse_iso_intervals.o ext/date/lib/.libs/interval.o ext/ereg/.libs/ereg.o ext/ereg/regex/.libs/regcomp.o ext/ereg/regex/.libs/regexec.o ext/ereg/regex/.libs/regerror.o ext/ereg/regex/.libs/regfree.o ext/libxml/.libs/libxml.o ext/openssl/.libs/openssl.o ext/openssl/.libs/xp_ssl.o ext/pcre/.libs/php_pcre.o ext/zlib/.libs/zlib.o ext/zlib/.libs/zlib_fopen_wrapper.o ext/zlib/.libs/zlib_filter.o ext/filter/.libs/filter.o ext/filter/.libs/sanitizing_filters.o ext/filter/.libs/logical_filters.o ext/filter/.libs/callback_filter.o ext/hash/.libs/hash.o ext/hash/.libs/hash_md.o ext/hash/.libs/hash_sha.o ext/hash/.libs/hash_ripemd.o ext/hash/.libs/hash_haval.o ext/hash/.libs/hash_tiger.o ext/hash/.libs/hash_gost.o ext/hash/.libs/hash_snefru.o ext/hash/.libs/hash_whirlpool.o ext/hash/.libs/hash_adler32.o ext/hash/.libs/hash_crc32.o ext/hash/.libs/hash_fnv.o ext/hash/.libs/hash_joaat.o ext/reflection/.libs/php_reflection.o ext/session/.libs/mod_user_class.o ext/session/.libs/session.o ext/session/.libs/mod_files.o ext/session/.libs/mod_mm.o ext/session/.libs/mod_user.o ext/spl/.libs/php_spl.o ext/spl/.libs/spl_functions.o ext/spl/.libs/spl_engine.o ext/spl/.libs/spl_iterators.o ext/spl/.libs/spl_array.o ext/spl/.libs/spl_directory.o ext/spl/.libs/spl_exceptions.o ext/spl/.libs/spl_observer.o ext/spl/.libs/spl_dllist.o ext/spl/.libs/spl_heap.o ext/spl/.libs/spl_fixedarray.o ext/standard/.libs/crypt_freesec.o ext/standard/.libs/crypt_blowfish.o ext/standard/.libs/crypt_sha512.o ext/standard/.libs/crypt_sha256.o ext/standard/.libs/php_crypt_r.o ext/standard/.libs/array.o ext/standard/.libs/base64.o ext/standard/.libs/basic_functions.o ext/standard/.libs/browscap.o ext/standard/.libs/crc32.o ext/standard/.libs/crypt.o ext/standard/.libs/cyr_convert.o ext/standard/.libs/datetime.o ext/standard/.libs/dir.o ext/standard/.libs/dl.o ext/standard/.libs/dns.o ext/standard/.libs/exec.o ext/standard/.libs/file.o ext/standard/.libs/filestat.o ext/standard/.libs/flock_compat.o ext/standard/.libs/formatted_print.o ext/standard/.libs/fsock.o ext/standard/.libs/head.o ext/standard/.libs/html.o ext/standard/.libs/image.o ext/standard/.libs/info.o ext/standard/.libs/iptc.o ext/standard/.libs/lcg.o ext/standard/.libs/link.o ext/standard/.libs/mail.o ext/standard/.libs/math.o ext/standard/.libs/md5.o ext/standard/.libs/metaphone.o ext/standard/.libs/microtime.o ext/standard/.libs/pack.o ext/standard/.libs/pageinfo.o ext/standard/.libs/quot_print.o ext/standard/.libs/rand.o ext/standard/.libs/soundex.o ext/standard/.libs/string.o ext/standard/.libs/scanf.o ext/standard/.libs/syslog.o ext/standard/.libs/type.o ext/standard/.libs/uniqid.o ext/standard/.libs/url.o ext/standard/.libs/var.o ext/standard/.libs/versioning.o ext/standard/.libs/assert.o ext/standard/.libs/strnatcmp.o ext/standard/.libs/levenshtein.o ext/standard/.libs/incomplete_class.o ext/standard/.libs/url_scanner_ex.o ext/standard/.libs/ftp_fopen_wrapper.o ext/standard/.libs/http_fopen_wrapper.o ext/standard/.libs/php_fopen_wrapper.o ext/standard/.libs/credits.o ext/standard/.libs/css.o ext/standard/.libs/var_unserializer.o ext/standard/.libs/ftok.o ext/standard/.libs/sha1.o ext/standard/.libs/user_filters.o ext/standard/.libs/uuencode.o ext/standard/.libs/filters.o ext/standard/.libs/proc_open.o ext/standard/.libs/streamsfuncs.o ext/standard/.libs/http.o ext/standard/.libs/password.o TSRM/.libs/TSRM.o TSRM/.libs/tsrm_strtok_r.o main/.libs/main.o main/.libs/snprintf.o main/.libs/spprintf.o main/.libs/php_sprintf.o main/.libs/fopen_wrappers.o main/.libs/alloca.o main/.libs/php_scandir.o main/.libs/php_ini.o main/.libs/SAPI.o main/.libs/rfc1867.o main/.libs/php_content_types.o main/.libs/strlcpy.o main/.libs/strlcat.o main/.libs/mergesort.o main/.libs/reentrancy.o main/.libs/php_variables.o main/.libs/php_ticks.o main/.libs/network.o main/.libs/php_open_temporary_file.o main/.libs/output.o main/.libs/getopt.o main/streams/.libs/streams.o main/streams/.libs/cast.o main/streams/.libs/memory.o main/streams/.libs/filter.o main/streams/.libs/plain_wrapper.o main/streams/.libs/userspace.o main/streams/.libs/transports.o main/streams/.libs/xp_socket.o main/streams/.libs/mmap.o main/streams/.libs/glob_wrapper.o Zend/.libs/zend_language_parser.o Zend/.libs/zend_language_scanner.o Zend/.libs/zend_ini_parser.o Zend/.libs/zend_ini_scanner.o Zend/.libs/zend_alloc.o Zend/.libs/zend_compile.o Zend/.libs/zend_constants.o Zend/.libs/zend_dynamic_array.o Zend/.libs/zend_dtrace.o Zend/.libs/zend_execute_API.o Zend/.libs/zend_highlight.o Zend/.libs/zend_llist.o Zend/.libs/zend_vm_opcodes.o Zend/.libs/zend_opcode.o Zend/.libs/zend_operators.o Zend/.libs/zend_ptr_stack.o Zend/.libs/zend_stack.o Zend/.libs/zend_variables.o Zend/.libs/zend.o Zend/.libs/zend_API.o Zend/.libs/zend_extensions.o Zend/.libs/zend_hash.o Zend/.libs/zend_list.o Zend/.libs/zend_indent.o Zend/.libs/zend_builtin_functions.o Zend/.libs/zend_sprintf.o Zend/.libs/zend_ini.o Zend/.libs/zend_qsort.o Zend/.libs/zend_multibyte.o Zend/.libs/zend_ts_hash.o Zend/.libs/zend_stream.o Zend/.libs/zend_iterators.o Zend/.libs/zend_interfaces.o Zend/.libs/zend_exceptions.o Zend/.libs/zend_strtod.o Zend/.libs/zend_gc.o Zend/.libs/zend_closures.o Zend/.libs/zend_float.o Zend/.libs/zend_string.o Zend/.libs/zend_signal.o Zend/.libs/zend_generators.o Zend/.libs/zend_virtual_cwd.o Zend/.libs/zend_ast.o Zend/.libs/zend_objects.o Zend/.libs/zend_object_handlers.o Zend/.libs/zend_objects_API.o Zend/.libs/zend_default_classes.o Zend/.libs/zend_execute.o main/.libs/internal_functions_cli.o sapi/cli/.libs/php_cli.o sapi/cli/.libs/php_http_parser.o sapi/cli/.libs/php_cli_server.o sapi/cli/.libs/ps_title.o sapi/cli/.libs/php_cli_process_title.o -o sapi/cli/php -Wl,--export-dynamic  -L/usr/lib/x86_64-linux-gnu/mit-krb5 -L/usr/lib/x86_64-linux-gnu -lcrypt -lresolv -lcrypt -lstdc++ -lz -lpcre -lrt -lm -ldl -lnsl -lxml2 -lgssapi_krb5 -lkrb5 -lk5crypto -lcom_err -lssl -lcrypto -lcrypt -lcrypt
/usr/bin/ld: Zend/.libs/zend_dtrace.o:./ext-build/./Zend/zend_dtrace.h:35: multiple definition of `zend_dtrace_execute_internal'; main/.libs/main.o:./ext-build/./Zend/zend_dtrace.h:35: first defined here
/usr/bin/ld: Zend/.libs/zend_dtrace.o:./ext-build/./Zend/zend_dtrace.h:34: multiple definition of `zend_dtrace_execute'; main/.libs/main.o:./ext-build/./Zend/zend_dtrace.h:34: first defined here
/usr/bin/ld: Zend/.libs/zend_dtrace.o:./ext-build/./Zend/zend_dtrace.h:33: multiple definition of `zend_dtrace_compile_file'; main/.libs/main.o:./ext-build/./Zend/zend_dtrace.h:33: first defined here
/usr/bin/ld: Zend/.libs/zend.o:./ext-build/./Zend/zend_dtrace.h:35: multiple definition of `zend_dtrace_execute_internal'; main/.libs/main.o:./ext-build/./Zend/zend_dtrace.h:35: first defined here
/usr/bin/ld: Zend/.libs/zend.o:./ext-build/./Zend/zend_dtrace.h:34: multiple definition of `zend_dtrace_execute'; main/.libs/main.o:./ext-build/./Zend/zend_dtrace.h:34: first defined here
/usr/bin/ld: Zend/.libs/zend.o:./ext-build/./Zend/zend_dtrace.h:33: multiple definition of `zend_dtrace_compile_file'; main/.libs/main.o:./ext-build/./Zend/zend_dtrace.h:33: first defined here
/usr/bin/ld: Zend/.libs/zend_exceptions.o:./ext-build/./Zend/zend_dtrace.h:35: multiple definition of `zend_dtrace_execute_internal'; main/.libs/main.o:./ext-build/./Zend/zend_dtrace.h:35: first defined here
/usr/bin/ld: Zend/.libs/zend_exceptions.o:./ext-build/./Zend/zend_dtrace.h:34: multiple definition of `zend_dtrace_execute'; main/.libs/main.o:./ext-build/./Zend/zend_dtrace.h:34: first defined here
/usr/bin/ld: Zend/.libs/zend_exceptions.o:./ext-build/./Zend/zend_dtrace.h:33: multiple definition of `zend_dtrace_compile_file'; main/.libs/main.o:./ext-build/./Zend/zend_dtrace.h:33: first defined here
/usr/bin/ld: Zend/.libs/zend_execute.o:./ext-build/./Zend/zend_dtrace.h:35: multiple definition of `zend_dtrace_execute_internal'; main/.libs/main.o:./ext-build/./Zend/zend_dtrace.h:35: first defined here
/usr/bin/ld: Zend/.libs/zend_execute.o:./ext-build/./Zend/zend_dtrace.h:34: multiple definition of `zend_dtrace_execute'; main/.libs/main.o:./ext-build/./Zend/zend_dtrace.h:34: first defined here
/usr/bin/ld: Zend/.libs/zend_execute.o:./ext-build/./Zend/zend_dtrace.h:33: multiple definition of `zend_dtrace_compile_file'; main/.libs/main.o:./ext-build/./Zend/zend_dtrace.h:33: first defined here
collect2: error: ld returned 1 exit status
make[2]: *** [Makefile:347: sapi/cli/php] Error 1
make[2]: Leaving directory '/usr/local/src/bugfix-debs/php/php5.6/php5.6-5.6.40/ext-build'
dh_auto_build: error: cd ext-build && make -j8 returned exit code 2
make[1]: *** [debian/rules:315: override_dh_auto_build-arch] Error 2
make[1]: Leaving directory '/usr/local/src/bugfix-debs/php/php5.6/php5.6-5.6.40'
make: *** [debian/rules:256: binary] Error 2
dpkg-buildpackage: error: fakeroot debian/rules binary subprocess returned exit status 2
```

**Distribution (please complete the following information):**
 - OS: Debian
 - Architecture: amd64
 - Repository: `deb-src https://packages.sury.org/php/ buster main`

Note I'm building myself on Debian unstable, as you don't offer binary packages for unstable. I'm not trying to argue that policy here, but at some point gcc 10 will make it into a stable release.

**Comment by LynxChaus** [comment]
add ```-fcommon``` to CFLAGS

**Comment by oerdnj** [comment]
So, you are building outdated software for the latest-greatest distribution... 🤯🤯🤯

I'll fix that it should be fairly easy fix...

**Comment by anomiex** [comment]
Yeah, I like to keep my own environment up to date, but still sometimes need to test with outdated PHP when working with projects that still support those versions.

**Comment by oerdnj** [comment]
Fixed in 0899b4cd65e5544cdcd4098b2032125411394aa5
